mpd: doesn't build on master

## Issue description

See the issue title.

### Steps to reproduce

```
$ nix-build -A mpd
...
g++ -DHAVE_CONFIG_H -I.  -I./src  -DSYSTEM_CONFIG_FILE_LOCATION='"/nix/store/g25wm18a3paqdamc2lzrap58jkg3pxw8-mpd-0.20.21/etc/mpd.conf"' -I/nix/store/8a02ivkcs7i251k6vb3mpbwqrbqmpzpi-libvorbis-1.3.6-dev/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include  -I/nix/store/7p6hqa237db4j7g7birnfn07bgjxjiqs-flac-1.3.2-dev/include  -I/nix/store/ziaz37psh1w02shqwdsz95gb0j25xjif-audiofile-0.3.6/include -I/nix/store/i7idy3j1pgb111g9y2ydvhxr7dhsckzc-libmikmod-3.3.11.1/include -pthread -D_REENTRANT -I/nix/store/i6rygb551z440bwf9z6k4d48fr4lpzap-game-music-emu-0.6.1/include  -I/nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include    -I/nix/store/m3p3kpdcv61j55c9smh0im424m0lxq5r-mpg123-1.25.10/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include -I/nix/store/j08difclns8145j4jjmif8b7hmjjm5vk-libopus-1.3-dev/include/opus -I/nix/store/4bpx8556255m3z0b18a4l1zb66d6cl2f-ffmpeg-3.4.4-dev/include    -D_GNU_SOURCE -pthread -g -O2 -fvisibility=hidden -fno-threadsafe-statics -fmerge-all-constants -ffast-math -ftree-vectorize -Wall -Wextra -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wwrite-strings -Wsign-compare -Wno-noexcept-type -c -o src/decoder/plugins/libdecoder_a-MikmodDecoderPlugin.o `test -f 'src/decoder/plugins/MikmodDecoderPlugin.cxx' || echo './'`src/decoder/plugins/MikmodDecoderPlugin.cxx                                                                                           
g++ -DHAVE_CONFIG_H -I.  -I./src  -DSYSTEM_CONFIG_FILE_LOCATION='"/nix/store/g25wm18a3paqdamc2lzrap58jkg3pxw8-mpd-0.20.21/etc/mpd.conf"' -I/nix/store/8a02ivkcs7i251k6vb3mpbwqrbqmpzpi-libvorbis-1.3.6-dev/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include  -I/nix/store/7p6hqa237db4j7g7birnfn07bgjxjiqs-flac-1.3.2-dev/include  -I/nix/store/ziaz37psh1w02shqwdsz95gb0j25xjif-audiofile-0.3.6/include -I/nix/store/i7idy3j1pgb111g9y2ydvhxr7dhsckzc-libmikmod-3.3.11.1/include -pthread -D_REENTRANT -I/nix/store/i6rygb551z440bwf9z6k4d48fr4lpzap-game-music-emu-0.6.1/include  -I/nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include    -I/nix/store/m3p3kpdcv61j55c9smh0im424m0lxq5r-mpg123-1.25.10/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include -I/nix/store/j08difclns8145j4jjmif8b7hmjjm5vk-libopus-1.3-dev/include/opus -I/nix/store/4bpx8556255m3z0b18a4l1zb66d6cl2f-ffmpeg-3.4.4-dev/include    -D_GNU_SOURCE -pthread -g -O2 -fvisibility=hidden -fno-threadsafe-statics -fmerge-all-constants -ffast-math -ftree-vectorize -Wall -Wextra -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wwrite-strings -Wsign-compare -Wno-noexcept-type -c -o src/decoder/plugins/libdecoder_a-FluidsynthDecoderPlugin.o `test -f 'src/decoder/plugins/FluidsynthDecoderPlugin.cxx' || echo './'`src/decoder/plugins/FluidsynthDecoderPlugin.cxx                                                                               
src/decoder/plugins/FluidsynthDecoderPlugin.cxx: In function 'bool fluidsynth_init(const ConfigBlock&)':
src/decoder/plugins/FluidsynthDecoderPlugin.cxx:84:47: error: invalid conversion from 'void (*)(int, char*, void*)' to 'fluid_log_function_t {aka void (*)(int, const char*, void*)}' [-fpermissive]             
           fluidsynth_mpd_log_function, nullptr);
                                               ^
In file included from /nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include/fluidsynth.h:105:0,                                                                                                   
                 from src/decoder/plugins/FluidsynthDecoderPlugin.cxx:29:
/nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include/fluidsynth/log.h:76:22: note:   initializing argument 2 of 'void (* fluid_set_log_function(int, fluid_log_function_t, void*))(int, const char*, void*)'
 fluid_log_function_t fluid_set_log_function(int level, fluid_log_function_t fun, void *data);
                      ^~~~~~~~~~~~~~~~~~~~~~
g++ -DHAVE_CONFIG_H -I.  -I./src  -DSYSTEM_CONFIG_FILE_LOCATION='"/nix/store/g25wm18a3paqdamc2lzrap58jkg3pxw8-mpd-0.20.21/etc/mpd.conf"' -I/nix/store/8a02ivkcs7i251k6vb3mpbwqrbqmpzpi-libvorbis-1.3.6-dev/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include  -I/nix/store/7p6hqa237db4j7g7birnfn07bgjxjiqs-flac-1.3.2-dev/include  -I/nix/store/ziaz37psh1w02shqwdsz95gb0j25xjif-audiofile-0.3.6/include -I/nix/store/i7idy3j1pgb111g9y2ydvhxr7dhsckzc-libmikmod-3.3.11.1/include -pthread -D_REENTRANT -I/nix/store/i6rygb551z440bwf9z6k4d48fr4lpzap-game-music-emu-0.6.1/include  -I/nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include    -I/nix/store/m3p3kpdcv61j55c9smh0im424m0lxq5r-mpg123-1.25.10/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include -I/nix/store/j08difclns8145j4jjmif8b7hmjjm5vk-libopus-1.3-dev/include/opus -I/nix/store/4bpx8556255m3z0b18a4l1zb66d6cl2f-ffmpeg-3.4.4-dev/include    -D_GNU_SOURCE -pthread -g -O2 -fvisibility=hidden -fno-threadsafe-statics -fmerge-all-constants -ffast-math -ftree-vectorize -Wall -Wextra -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wwrite-strings -Wsign-compare -Wno-noexcept-type -c -o src/decoder/plugins/libdecoder_a-FfmpegIo.o `test -f 'src/decoder/plugins/FfmpegIo.cxx' || echo './'`src/decoder/plugins/FfmpegIo.cxx
g++ -DHAVE_CONFIG_H -I.  -I./src  -DSYSTEM_CONFIG_FILE_LOCATION='"/nix/store/g25wm18a3paqdamc2lzrap58jkg3pxw8-mpd-0.20.21/etc/mpd.conf"' -I/nix/store/8a02ivkcs7i251k6vb3mpbwqrbqmpzpi-libvorbis-1.3.6-dev/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include  -I/nix/store/7p6hqa237db4j7g7birnfn07bgjxjiqs-flac-1.3.2-dev/include  -I/nix/store/ziaz37psh1w02shqwdsz95gb0j25xjif-audiofile-0.3.6/include -I/nix/store/i7idy3j1pgb111g9y2ydvhxr7dhsckzc-libmikmod-3.3.11.1/include -pthread -D_REENTRANT -I/nix/store/i6rygb551z440bwf9z6k4d48fr4lpzap-game-music-emu-0.6.1/include  -I/nix/store/q15xhik25xsdlnq8v73nkl36cxqhf76v-fluidsynth-2.0.1/include    -I/nix/store/m3p3kpdcv61j55c9smh0im424m0lxq5r-mpg123-1.25.10/include -I/nix/store/342z6rg6m1ahk8rcnrhlm1yqnv2kr4cr-libogg-1.3.3-dev/include -I/nix/store/j08difclns8145j4jjmif8b7hmjjm5vk-libopus-1.3-dev/include/opus -I/nix/store/4bpx8556255m3z0b18a4l1zb66d6cl2f-ffmpeg-3.4.4-dev/include    -D_GNU_SOURCE -pthread -g -O2 -fvisibility=hidden -fno-threadsafe-statics -fmerge-all-constants -ffast-math -ftree-vectorize -Wall -Wextra -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wwrite-strings -Wsign-compare -Wno-noexcept-type -c -o src/decoder/plugins/libdecoder_a-FfmpegMetaData.o `test -f 'src/decoder/plugins/FfmpegMetaData.cxx' || echo './'`src/decoder/plugins/FfmpegMetaData.cxx
make[1]: *** [Makefile:9419: src/decoder/plugins/libdecoder_a-FluidsynthDecoderPlugin.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/source'
make: *** [Makefile:5337: all] Error 2
builder for '/nix/store/axcacxfavqk7f5cghvv1lcggv3h9yxz1-mpd-0.20.21.drv' failed with exit code 2
error: build of '/nix/store/axcacxfavqk7f5cghvv1lcggv3h9yxz1-mpd-0.20.21.drv' failed
```


## Technical details

nixpkgs @ a296b6eda746c4e06986cb961a384f6fc99ed449

--- a/pkgs/servers/mpd/default.nix
+++ b/pkgs/servers/mpd/default.nix
@@ -36,7 +36,7 @@ let
   opt = stdenv.lib.optional;
   mkFlag = c: f: if c then "--enable-${f}" else "--disable-${f}";
   major = "0.20";
-  minor = "21";
+  minor = "23";
 
 in stdenv.mkDerivation rec {
   name = "mpd-${version}";
@@ -46,7 +46,7 @@ in stdenv.mkDerivation rec {
     owner  = "MusicPlayerDaemon";
     repo   = "MPD";
     rev    = "v${version}";
-    sha256 = "0qchvycwiai5gwkvvf44nc1jw16yhpcjmlppqlrlvicgzsanhmy3";
+    sha256 = "1z1pdgiddimnmck0ardrpxkvgk1wn9zxri5wfv5ppasbb7kfm350";
   };
 
   patches = [ ./x86.patch ];

